stainless_frontend/tests/pass/clone_equality.rs

```diff
@@ -1,23 +1,27 @@
 extern crate stainless;
 use stainless::*;
 
-trait EqualClone: Sized {
-  fn clone(&self) -> Self;
-
+trait Equals {
   fn eq(&self, other: &Self) -> bool;
+}
 
-  #[law]
-  fn preserve_equality(a: &Self, b: &Self) -> bool {
-    (a.eq(b) == a.clone().eq(b)) == (a.eq(&b.clone()) == a.clone().eq(&b.clone()))
-  }
+trait Clone {
+  fn clone(&self) -> Self;
 }
 
+#[external]
+#[pure]
+#[post(x.clone().eq(&x))]
+#[allow(unused_variables)]
+fn clone_preserves_eq<T: Clone + Equals>(x: T) {}
```

Comment:
This axiom is not used anywhere, let's remove it?

Reply:
This is just what was failing before the PR, but I can put it in a separate test.